cache/cache.go

```diff
@@ -226,3 +230,59 @@ func readWatchChannel(
 	}
 	return nil
 }
+
+func (c *Cache) validateWatch(key string, opts ...clientv3.OpOption) (pred KeyPredicate, err error) {
+	op := clientv3.OpGet(key, opts...)
+	startRev := op.Rev()
+	// TODO: Support watch on arbitrary startRev support once we guarantee gap-free replay.
+	if startRev != 0 {
+		return nil, ErrUnsupportedWatch
+	}
+
+	startKey := []byte(key)
+	endKey := op.RangeBytes() // nil = single key, {0}=FromKey, else explicit range
+
+	if err := c.validateWatchRange(startKey, endKey); err != nil {
+		return nil, err
+	}
+	return KeyPredForRange(startKey, endKey), nil
+}
+
+func (c *Cache) validateWatchRange(startKey, endKey []byte) error {
+	isFrom := len(endKey) == 1 && endKey[0] == 0
+	isRange := endKey != nil && !isFrom
+	prefixStart := []byte(c.prefix)
+	prefixEnd := []byte(clientv3.GetPrefixRangeEnd(c.prefix))
+
+	if isRange && bytes.Compare(endKey, startKey) <= 0 {
+		return ErrUnsupportedWatch
+	}
+
+	if c.prefix == "" {
```

Comment:
Don't like that you are treating `c.prefix == ""` as special case. Will rest of the validation fail?